### DISCUSSION TITLE
Unable to save changes

### DISCUSSION THREAD

**Discussion by vk3il** [opengisch/QField#4903] [Bugs]
I get the error "unable to save changes" whenever I try to add a new feature to a map on any layer in my QFieldCloud project. This was previously working fine, but am now unable to save anything. I have tried re-installing QField and re-downloading the project from QFieldCloud and the error is consistently present.

Looking at the log, there are the following notable errors: "Provider errors: wrong data type for attribute 1 of feature -6: 10", "Unbalanced call to leaveUpdateMode() w.r.t. enterUpdateMode()" and "Could not save changes. Rolling back."

Any help is appreciated.

David.

**Comment by suricactus** [comment]
Can you please share which QField version you are using?

**Comment by vk3il** [comment]
Sure. It's version 3.1.4- Borneo (d95c92 V3.1.4) using Qt 6.6.1.

**Comment by suricactus** [comment]
Thank you for sharing. I have tried with equivalent of the demo bees project and failed to reproduce. Can you share a minimal version of your project that reproduces the error?

**Reply by vk3il** [reply]
@suricactus Thanks, I've tried to create a minimal version, but unable to reproduce other than the full project as not sure what the problem is. Probably easiest if I share the whole project (it's not too large). I've added you as a collaborator on the project on Qfield Cloud - hopefully that gives you the access needed.

**Comment by m-kuhn** [comment]
Could be related to https://github.com/opengisch/QField/issues/4909

**Reply by vk3il** [reply]
Yes, this looks like the same or a very similar issue. Completing a set of attributes and then being unable to save. In my case I am using Android 14 on a Samsung Galaxy S23+.

**Reply by suricactus** [reply]
@vk3il did you have a look at this answer https://github.com/opengisch/QField/issues/4909#issuecomment-1890259290 ? Is it the same issue?

**Reply by vk3il** [reply]
Possibly. The only potential type mismatch I can see is the fid field which is Integer, which I populate automatically using uuid(). What I'm not clear on is the type of uuid(). Is it string or integer?

This has worked reliably in previous versions of QField and QGIS doesn't seem to mind it. If this is the problem, what is the recommended way to automatically populate fid fields?

**Reply by suricactus** [reply]
`uuid()`  returns a string. 

If it recently broke, can you give us a hint what was the last version it worked correctly?

**Reply by vk3il** [reply]
OK, the fid type mismatch seems to have been the problem I can now create features OK. I have replaced uuid() with @id as the default for the field which seems to mostly work. The only problem using @id is that the first new feature created in QField seems to get a very large negative number (-9223372036854775808 or 0x8000000000000000) for @id which then breaks QGIS Identify features tool (it highlights the feature, but won't pop up the information panel). Subsequent features created on the same layer get reasonable values assigned for fid.

I can't be sure which version of QField it previously worked in, but the last successful recording of a feature was 20 November 2023, so whichever was the current version at that time worked OK. 

Thanks for your assistance.

**Reply by vk3il** [reply]
Just a little more experimenting with feature creation. The @id problem seems to occur again if the first feature created (with the incorrect fid) is deleted. The next feature created then receives the very large negative value for fid from @id. Seems to maybe be an initialisation problem with the @id value?
